alacritty_terminal/src/term/mod.rs

```diff
@@ -382,7 +387,7 @@ impl<T> Term<T> {
             res += &self.line_to_string(line, start_col..end_col, line == end.line);
         }
 
-        res
+        res.strip_suffix('\n').map(str::to_owned).unwrap_or(res)
```

Comment:
Why wouldn't we do this in the `line_to_string` function? Currently we add newlines conditionally there just to strip it here. Which seems rather… pointless.

Alacritty will never have `\n` in any cell, the only way there could ever be one is if it's inserted by `line_to_string` because of the last column.

Reply:
Um... now Alacritty implicitly uses a trailing newline character of the string returned from `line_to_string` to indicate whether or not the range contains the last column or the end of a line is not wrapped. This is useful to join strings correctly in `bounds_to_string`, especially for wrapline joining. So we would need to treat wrapline by `bounds_to_string` instead of `line_to_string` if we trim a newline character by `line_to_string`. We need to lookup the grid contents to treat wrapline, but I think it is better to do it by `line_to_string` for abstraction.

Reply:
Oh yeah wrapline makes sense.